hack/kind/setup.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+set -e -u -o pipefail
+
+declare -r SCRIPT_PATH=$(readlink -f "$0")
+declare -r SCRIPT_DIR=$(cd $(dirname "$SCRIPT_PATH") && pwd)
+declare -r PROJECT_ROOT_DIR="$SCRIPT_DIR/../../"
+declare -r OP_NAME="obs-operator"
+
+### create a separate config file for kind clusters
+export KUBECONFIG=~/.kube/kind/obs-operator
+
+err() {
+  echo -e "ERROR: $@" >&2
+}
+
+info() {
+  echo -e "INFO: $@"
+}
+
+die() {
+  echo -e "FATAL: $@" >&2
+  exit 1
+}
+
+# turn the control-plane into to infra to validate if the operator pods
+# get deployed on infra nodes
+label_infra_node() {
+  kubectl label nodes  \
+    -l kubernetes.io/hostname=="${OP_NAME}-control-plane"\
+    node-role.kubernetes.io/infra="" \
+
+  [[ $( kubectl get nodes -l node-role.kubernetes.io/infra=="" -o name | wc -l ) -ne 0 ]] || {
+    die "No infra nodes were found"
+  }
+}
+
+setup_olm() {
+  [[ -x $PROJECT_ROOT_DIR/tmp/bin/operator-sdk  ]] || {
+    die "operator-sdk not found - did you run 'make tools'?"
+  }
+  $PROJECT_ROOT_DIR/tmp/bin/operator-sdk olm install
+  echo -e "      ---------------------------------- \n"
+}
+
+run_registry() {
+  info "Deploying Registry ..."
+
+  grep -q local-registry /etc/hosts || {
+    err "No local-registry entry in hosts; try:"
+    info "echo \"127.0.0.1\tlocal-registry\" | sudo tee -a /etc/hosts"
+    die "/etc/hosts does not contain local-registry entry"
+  }
+
+  kubectl apply -f ./hack/kind/registry.yaml -n operators
+  kubectl rollout status deployment local-registry -n operators
+
+  kubectl wait --for=condition=Available deploy local-registry -n operators --timeout=300s
+
+  curl http://local-registry:30000 || {
```

Comment:
In my machine, this step actually failed. I commented out the steps before `run-registry` and re-ran the script and it worked. So I feel like here we should actually retry the connection for a couple of sec/min.

Reply:
I am surprised why this happens .. The deployment did say it is available (see above) . 
I will add a return / max-timeout to curl.

Reply:
Tried it again and it ran perfectly 👌